build.gradle.kts

```diff
@@ -0,0 +1,372 @@
+        }
+    }
+
+    tasks.test {
+        configure<JacocoTaskExtension> {
+            // don't instrument sdk, icons, ktlint, etc.
+            includes = listOf("software.aws.toolkits.*")
+            excludes = listOf("software.aws.toolkits.ktlint.*")
+        }
+
+        reports {
+            junitXml.isEnabled = true
+            html.isEnabled = true
+        }
+    }
+
+    plugins.withType<IdeaPlugin> {
+        model.module.apply {
+            sourceDirs.plusAssign(sourceSets.main.get().java.srcDirs)
+            resourceDirs.plusAssign(sourceSets.main.get().resources.srcDirs)
+            testSourceDirs.plusAssign(File("tst-$ideVersion"))
+            testResourceDirs.plusAssign(File("tst-resources-$ideVersion"))
+
+            sourceDirs.minusAssign(File("it"))
+            testSourceDirs.plusAssign(File("it"))
+            testSourceDirs.plusAssign(File("it-$ideVersion"))
+
+            resourceDirs.minusAssign(File("it-resources"))
+            testResourceDirs.plusAssign(File("it-resources"))
+            testResourceDirs.plusAssign(File("it-resources-$ideVersion"))
+        }
+    }
+
+    tasks.register<Test>("integrationTest") {
+        group = LifecycleBasePlugin.VERIFICATION_GROUP
+        description = "Runs the integration tests."
+        testClassesDirs = sourceSets["integrationTest"].output.classesDirs
+        classpath = sourceSets["integrationTest"].runtimeClasspath
+
+        configure<JacocoTaskExtension> {
+            // don"t instrument sdk, icons, ktlint, etc.
+            includes = listOf("software.aws.toolkits.*")
+            excludes = listOf("software.aws.toolkits.ktlint.*")
+        }
+
+        project.plugins.withId("org.jetbrains.intellij") {
+            systemProperty("log.dir", "${(project.extensions["intellij"] as org.jetbrains.intellij.IntelliJPluginExtension).sandboxDirectory}-test/logs")
+        }
+
+        mustRunAfter(tasks.test)
+    }
+
+    project.plugins.withId("org.jetbrains.intellij") {
+        tasks.withType(org.jetbrains.intellij.tasks.DownloadRobotServerPluginTask::class.java) {
+            this.version = remoteRobotVersion
+        }
+
+        tasks.withType(org.jetbrains.intellij.tasks.RunIdeForUiTestTask::class.java).all {
+            systemProperty("robot-server.port", remoteRobotPort)
+            systemProperty("ide.mac.file.chooser.native", "false")
+            systemProperty("jb.consents.confirmation.enabled", "false")
+            // This does some magic in EndUserAgreement.java to make it not show the privacy policy
+            systemProperty("jb.privacy.policy.text", "<!--999.999-->")
+            if (System.getenv("CI") != null) {
+                systemProperty("aws.sharedCredentialsFile", "/tmp/.aws/credentials")
+            }
+        }
+
+        extensions.getByType<JacocoPluginExtension>().applyTo(tasks.getByName<RunIdeForUiTestTask>("runIdeForUiTests"))
+    }
+
+    tasks.withType(KotlinCompile::class.java).all {
```

Comment:
prefer the `<>` style over using the `::class.java`

Reply:
Agree, changed